Build/SaveLoadScreen.cc

```diff
@@ -1731,12 +1731,11 @@ void DoDeadIsDeadSave()
 		ScreenID tmpGuiPreviousOptionScreen = guiPreviousOptionScreen;
 		// We want to save the current screen we are in. Unless we are in Options, Laptop, or others
 		// Make sure we are always in a sane screen.
-		if (guiCurrentScreen != MAP_SCREEN && guiCurrentScreen != GAME_SCREEN && tmpGuiPreviousOptionScreen != MAP_SCREEN && tmpGuiPreviousOptionScreen != GAME_SCREEN)
+		if (guiCurrentScreen != MAP_SCREEN && guiCurrentScreen != GAME_SCREEN)
```

Comment:
What you actually want to do here is:
`if (tmpGuiPreviousOptionScreen != MAP_SCREEN && tmpGuiPreviousOptionScreen != GAME_SCREEN)
{
	if(guiCurrentScreen != MAP_SCREEN && guiCurrentScreen != GAME_SCREEN )
	{
		// If all fails, go to the map screen, this (almost) guarantees the game will start
		guiPreviousOptionScreen = MAP_SCREEN;
	} else {
		guiPreviousOptionScreen = guiCurrentScreen;
	}
}`

The reason is that guiPreviousOptionsScreen is the one that is written to the savegame. By checking guiCurrentScreen as well (which can contain a valid Screen in situations where guiPreviousOptionsScreen does not) you increase the chance of getting the player back to where he/she was.